packages/optimistic-oracle/src/proposer.js

```diff
@@ -296,8 +296,86 @@ class OptimisticOracleProposer {
     }
   }
 
+  // Settle disputes where this bot was the disputer and proposals where this bot was the proposer.
   async settleRequests() {
-    // TODO
+    this.logger.debug({
+      at: "OptimisticOracleProposer#settleRequests",
+      message: "Checking for proposals and disputes to settle"
+    });
+
+    for (let priceRequest of this.optimisticOracleClient
+      .getSettleableProposals(this.account)
+      .concat(this.optimisticOracleClient.getSettleableDisputes(this.account))) {
+      // Create the transaction.
+      const settle = this.ooContract.methods.settle(
```

Comment:
this looks good to me, the only nit i have here is that id like to see a function which takes in a single priceRequest and has this logic, rather than it being in a loop. then the loop can call that function

Reply:
Good feedback, I refactored this method and `sendDisputes()` and `sendProposals()`